streams/src/main/java/org/apache/kafka/streams/kstream/internals/AbstractStream.java

```diff
@@ -31,6 +31,8 @@
     protected final Set<String> sourceNodes;
 
     public AbstractStream(KStreamBuilder topology, String name, Set<String> sourceNodes) {
+        assert sourceNodes != null && !sourceNodes.isEmpty() : "parameter <sourceNodes> must not be null or empty";
```

Comment:
Good point, and also I agree with you that we do not want to add verbose unit tests with overlapping coverage. In fact we already have quite a few in our current unit tests, especially for example kafka-streams / connect that have targeted module / functions which is already covered in kafka-clients. Let's keep the unit test as is.